### ISSUE TITLE
508-defect-2: Military service history needs design review (and possible redesign)

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#48261]
## What happened?
The military service history needs a design review as:
- [it uses a typeahead instead of the standard open text field](https://design.va.gov/patterns/ask-users-for/service-history)
- it provides 50+ possible selections despite [EVSS planning to only accept 23 in the future](https://dsva.slack.com/archives/C8R3JS8BU/p1649702827450209?thread_ts=1649700446.433219&cid=C8R3JS8BU)
- it's missing headings on added service history items
- it's missing unique labels for list loop buttons

For more on [a11y problems with the list loop pattern, visit this ticket from coe](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45566).

https://user-images.githubusercontent.com/14154792/195929366-db11f5e8-4152-468f-a786-50a2e711109a.mov

<img width="511" alt="Screen Shot 2022-10-14 at 3 48 15 PM" src="https://user-images.githubusercontent.com/14154792/195929779-69f7fc94-2a56-4941-bb48-07372797e6f8.png">

## Steps to Reproduce
- [Use the unsigned in version of pre-need](https://www.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/sponsor-military-history)
- Select "other" for relationship to service member
- Reach the step military service and confirm that the sponsor's branch of service question has 50+ options

## Desired behavior
- Uncertain what the desired behavior is. We should conduct a little secondary research to understand this better and figure out next steps.

## Acceptance Criteria
- [ ] Designer reviews military service history page
- [ ] ??

## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with `Bug`**

**Comment by ccruzgra** [comment]
@joshkimux In the desired Behavior where you state secondary research needs to be conducted to figure out what needs to be happening, is that Platform folks that need to do that or does that fall on my team?

**Comment by joshkimux** [comment]
@ccruzgra this would fall within your team 🙇

**Comment by MichelleDieudonne** [comment]
Jira work item link: https://vajira.max.gov/browse/MBMS-37682

**Comment by ccruzgra** [comment]
@coforma-terry Could you please clarify the issue above for `it's missing unique labels for list loop buttons`. Is that referring to the aria labels missing from the cards?

**Comment by eli-oat** [comment]
@ccruzgra thanks for this question! 

[this ticket provides more details about the list loop buttons](https://github.com/department-of-veterans-affairs/va.gov-team/issues/45566). I am new to this specific ticket, so am not certain of the markup you are working with (is there a link to this in staging I could review?), but the tl;dr is that when building a list of distinct but similar data you need someway to distinguish between the duplicate date other than visually. Typically this is resolved with aria, but it could be done with heading grouping, I believe (I'd want to confirm that with DST before suggesting that as a diffinitive answer).

**Comment by ccruzgra** [comment]
Hi @eli-oat. This form is found at https://staging.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/introduction.

**Comment by barron-charles** [comment]
@eli-oat did you have a chance to speak with DST regarding this issue?

**Comment by eli-oat** [comment]
I haven't yet spoken with them, @barron-charles 

- [multiple responses](https://design.va.gov/patterns/ask-users-for/multiple-responses)
- [VA Forms Library - How to work with Array Data (aka List Loops)](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-how-to-work-with-array-data-aka-l)

Sorry, I was speaking in the hypothetical and wasn't sure how you all wanted to proceed. I will reach out to them now and follow up.

**Comment by barron-charles** [comment]
Thanks you! It's much appreciated.

**Comment by eli-oat** [comment]
This has been a difficult one to determine definitively since there are a couple of list loop patterns at exist, including: 

- [Contact info list loop](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-how-to-create-the-contact-info-ar)
- [summary + add page list loop](https://depo-platform-documentation.scrollhelp.site/developer-docs/va-forms-library-how-to-use-add-item-link-in-array) which is a lot like contact info, but there is only 1 add/edit page
- [List loop + summary](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1697) where the summary is shown after adding everything. The docs for this one are wicked minimal

That said, the issue initially flagged is still present, wherein you cannot clearly navigate items in any way other than visually. 

The heading map shows this, too: 

```
Apply Online For Pre-Need Determination Of Eligibility In A VA National Cemetery | Veterans Affairs
1 Apply for pre-need eligibility determination
    2 Step 2 of 6: Military historyWe’ll save your application on every change.
        3 New service period
    2 Need help?
    2 Veteran programs and services
    2 More VA resources
    2 Get VA updates
    2 In crisis? Talk to someone now
    2 Get answers
    2 Call us
    2 Visit a medical center or regional office
    2 Language assistance
```
<img width="784" alt="Screenshot 2023-09-19 at 10 44 03 AM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/3342274/469bb7aa-6d24-4038-baf0-52593ea3ceab">

When the heading map is compared to that screenshot of the form, you'll notice that the previous military history item isn't present -- this is similar to the experience in voice over, where it is difficult to navigate to the existing military history.
